### ISSUE TITLE
Better design for RandomSizedBBoxSafeCrop without implicit resizing

### ISSUE THREAD

**Issue by SunQpark** [albumentations-team/albumentations#480]
This is current definition of `RandomSizedBBoxSafeCrop` class, which is on the `transforms.py`
```python3
class RandomSizedBBoxSafeCrop(DualTransform):
    """Crop a random part of the input and rescale it to some size without loss of bboxes.

    Args:
        height (int): height after crop and resize.
        width (int): width after crop and resize.
        erosion_rate (float): erosion rate applied on input image height before crop.
        interpolation (OpenCV flag): flag that is used to specify the interpolation algorithm. Should be one of:
            cv2.INTER_NEAREST, cv2.INTER_LINEAR, cv2.INTER_CUBIC, cv2.INTER_AREA, cv2.INTER_LANCZOS4.
            Default: cv2.INTER_LINEAR.
        p (float): probability of applying the transform. Default: 1.

    Targets:
        image, mask, bboxes

    Image types:
        uint8, float32
    """

    def __init__(self, height, width, erosion_rate=0.0, interpolation=cv2.INTER_LINEAR, always_apply=False, p=1.0):
        super(RandomSizedBBoxSafeCrop, self).__init__(always_apply, p)
        self.height = height
        self.width = width
        self.interpolation = interpolation
        self.erosion_rate = erosion_rate

    def apply(self, img, crop_height=0, crop_width=0, h_start=0, w_start=0, interpolation=cv2.INTER_LINEAR, **params):
        crop = F.random_crop(img, crop_height, crop_width, h_start, w_start)
        return F.resize(crop, self.height, self.width, interpolation)
```
And this is my customization on the same class, without implicit resizing.
```python3
class RandomSizedBBoxSafeCrop(DualTransform):
    """Crop a random part of the input and rescale it to some size without loss of bboxes.

    Args:
        erosion_rate (float): erosion rate applied on input image height before crop.
        p (float): probability of applying the transform. Default: 1.

    Targets:
        image, mask, bboxes

    Image types:
        uint8, float32
    """

    def __init__(self, erosion_rate=0.0, always_apply=False, p=1.0):
        super(RandomSizedBBoxSafeCrop, self).__init__(always_apply, p)
        self.erosion_rate = erosion_rate

    def apply(self, img, crop_height=0, crop_width=0, h_start=0, w_start=0, **params):
        return F.random_crop(img, crop_height, crop_width, h_start, w_start)

```
I think the latter is better design, since it has simpler code and fits into more general use cases. If the resizing feature is really need for someone, it can be done explicitly by composing this with Resize.

**Comment by SunQpark** [comment]
Issue #298 seems to be related to this.
I tried to make this change in a PR, but it failed some test due to change of class signature.

**Comment by Dipet** [comment]
Yes current implementation has a little bit overhead, but we cannot broke backward compatibility for existing transforms. In next transforms we will try to create much simpler and brief code.

**Comment by Dipet** [comment]
Also this transform named such `Sized`, that mean it will be resized to some `height` and `width`. If you need safe crop without resize you can add add new transform `RandomSizedBBoxSafeCrop` with you signature.

**Comment by Dipet** [comment]
Also we can set `height` and `width` to `None` by default . And if one of it is `None` do not use resize

**Comment by SunQpark** [comment]
I'll soon make a PR with a new transform named `BBoxSafeRandomCrop` which does not resize output images as implementation posted above. 

With this new transform, I think the original implementation `RandomSizedBBoxSafeCrop` should be replaced with a much simpler implementation like below.

```python
class RandomSizedBBoxSafeCrop(BBoxSafeRandomCrop):
    """class docstring is omitted"""

    def __init__(self, height, width, erosion_rate=0.0, interpolation=cv2.INTER_LINEAR, always_apply=False, p=1.0):
        super(RandomSizedBBoxSafeCrop, self).__init__(erosion_rate, always_apply, p)
        self.height = height
        self.width = width
        self.interpolation = interpolation

    def apply(self, img, crop_height=0, crop_width=0, h_start=0, w_start=0, interpolation=cv2.INTER_LINEAR, **params):
        crop = super(RandomSizedBBoxSafeCrop, self).apply(img, crop_height=0, crop_width=0, h_start=0, w_start=0, interpolation=cv2.INTER_LINEAR, **params)
        return F.resize(crop, self.height, self.width, interpolation)
```
In this way, we can have clean and simple code without duplicated parts, while not breaking the backward compatibility.

**Comment by Dipet** [comment]
Implemented in #579
